tests/test_workspace_dry_clone.py

```
@@ -0,0 +1,113 @@
+        raise asyncio.TimeoutError()
+
+    monkeypatch.setattr(asyncio, 'wait_for', fake_wait_for)
+    r = await dry_clone_repo('https://github.com/example/repo.git', None, timeout=1)
+    assert r['ok'] is False
+    assert r['error'] == 'timeout'
+
+
+@pytest.mark.asyncio
+async def test_dry_clone_repo_sanitizes_token_in_error(monkeypatch):
+    """Token value is replaced with [REDACTED] in error message when subprocess fails."""
+    token = "ghp_SECRET_TOKEN_XYZ"
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to: "token"

(S105)